src/Octoshift/Services/AdoApi.cs

```diff
@@ -593,17 +597,512 @@ public virtual async Task ChangePipelineRepo(string adoOrg, string teamProject,
+            .Cast<JObject>()
+            .FirstOrDefault();
+
+        if (matchingTrigger != null)
+        {
+            // Return the original reportBuildStatus value, defaulting to "true" if not present
+            var reportBuildStatusToken = matchingTrigger["reportBuildStatus"];
+            if (reportBuildStatusToken == null)
+            {
+                return "true"; // Default to "true" when property doesn't exist
+            }
+
+            // Handle both boolean and string values - normalize to string
+            if (reportBuildStatusToken.Type == JTokenType.Boolean)
+            {
+                return reportBuildStatusToken.Value<bool>() ? "true" : "false";
+            }
+            else if (reportBuildStatusToken.Type == JTokenType.String)
+            {
+                var stringValue = reportBuildStatusToken.ToString();
+                // Normalize to lowercase for consistency
+                return string.Equals(stringValue, "true", StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(stringValue, "True", StringComparison.OrdinalIgnoreCase)
+                    ? "true"
+                    : "false";
+            }
+
+            // Try to convert any other type to boolean, then to string
+            try
+            {
+                return reportBuildStatusToken.ToObject<bool>() ? "true" : "false";
+            }
+            catch (JsonException)
+            {
+                return "true"; // Default to "true" if JSON conversion fails
+            }
+            catch (InvalidOperationException)
+            {
+                return "true"; // Default to "true" if operation is invalid
+            }
+        }
+
+        return "true"; // Default to "true" when trigger type not found in original triggers
+    }
+
+    private static bool IsReportBuildStatusEnabled(string reportBuildStatusValue)
+    {
+        return string.Equals(reportBuildStatusValue, "true", StringComparison.OrdinalIgnoreCase);
+    }
+
+    private bool HasTriggerType(JToken originalTriggers, string triggerType)
+    {
+        if (originalTriggers is not JArray triggerArray)
+        {
+            return false;
+        }
+
+        // Check if the specified trigger type exists
+        return triggerArray
+            .OfType<JObject>()
+            .Where(triggerObj => triggerObj["triggerType"]?.ToString() == triggerType)
+            .Any();
+    }
+
+    private async Task<(bool isRequired, bool checkSucceeded)> CheckBranchPolicyRequirement(string adoOrg, string teamProject, string currentRepoName, int pipelineId)
+    {
+        if (string.IsNullOrEmpty(currentRepoName))
+        {
+            return (false, false);
+        }
+
+        try
+        {
+            var isRequired = await IsPipelineRequiredByBranchPolicy(adoOrg, teamProject, currentRepoName, pipelineId);
+            return (isRequired, true);
+        }
+        catch (HttpRequestException)
```

Comment:
I wonder about these false cases. Is there some way to log the result in a migration report? The implication of all these catches is the resulting migrated repo will be silently missing branch policies compared to the source repo which will be surprising and hard to diagnose if now the vast majority of repos have the correct policies set post migration (as opposed to today where you at least now you have to fix this yourself in 100% of repos).